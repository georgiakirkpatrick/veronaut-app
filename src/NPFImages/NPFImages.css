.NPFImages .FormPromptWithSub {
    margin-bottom: 3vh;
}
/* 
.NPFImages__fieldset {
    position: relative;
    width: 100%;
    padding: 10px;
    margin: 5px 0;
    background-color: rgb(238, 238, 238);
    border: none;
}

.NPFImages__fieldset .FormPromptWithSub {
    margin-bottom: 2vh;
} */
.NPFImages__image-field {
    position: relative;
    width: 100%;
    padding: 0;
    margin: 1vh 0;
}

.NPFImages__image-field .NPFImages__remove-image {
    position: absolute;
    right: 0;
    padding: 0;
    background-color: #F2E7F1;
    color: #000;
    font: 500 14px/18px Poppins, sans-serif;
}

.NPFImages__preview-images {
    width: 100%;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: flex-start;
}

.NPFImages__preview-img { 
    width: 23%;
    margin-right: 2%
}