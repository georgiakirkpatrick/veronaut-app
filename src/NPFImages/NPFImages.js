import React from 'react'
import '@fortawesome/react-fontawesome'
import FormPage from '../FormPage/FormPage'
import FormButton from '../FormButton/FormButton'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import NPFFooter from '../NPFFooter/NPFFooter';

const NPFImages = props => {
    const {
        colorFieldsets,
        currentPage,
        setColorFieldsets,
        setPage
    } = props

    const addImage = (index) => {
        const fieldsetToUpdate = colorFieldsets[index]
        fieldsetToUpdate.imageUrls.push('')
        setColorFieldsets([ ...colorFieldsets ])    
    }

    const changeSwatchInput = (event, index) => {
        const values = [...colorFieldsets]
        console.log('values', values)

        values[index][event.target.name] = event.target.value
        console.log('values[index][event.target.name]', values[index][event.target.name])

        setColorFieldsets(values)
    }

    const changeImageInput = (index, i, event) => {
        console.log('index', index)
        console.log('i', i)
        console.log('event', event)

        const values = [...colorFieldsets]
    
        values[index][event.target.name][i] = event.target.value
    }

    const nextButton = () => {
        const missingSwatches = []
        const missingUrls = []

        colorFieldsets.forEach(fieldset => {
            if (fieldset.swatchUrl === '' || null) {
                missingSwatches.push(fieldset.name)
            }

            fieldset.imageUrls.forEach(field => {
                if (field === '' || null) {
                    missingUrls.push(
                        field.name
                    )
                }
            })
        })
        
        if (missingSwatches.length === 1) {
            alert(`Please complete the '${missingSwatches[0]}' swatch url field`)
        } else if (missingSwatches.length > 1) {
            alert(`Please complete the following fields: ${missingSwatches.map(field => `
                '${field}' swatch url`)}
            `)
        } else if (missingSwatches.length === 0) {
            if (missingUrls.length >= 1) {
                alert(`Please enter at least one image url for each color and remove any empty fields`)
            } else {
                setPage(currentPage + 1)
            }
        }
    }

    // const nextButton = () => {setPage(currentPage + 1)}

    const removeImage = (index, i) => {
        const newColors = [...colorFieldsets]
        newColors[index].imageUrls.splice(i, 1)
        setColorFieldsets(newColors)
    }

    return (
        <div>
            <FormPage title='Images'>
                <FormPromptWithSub 
                    prompt='Enter the color swatch URL and image URL(s) for each color'
                    promptSubtitle=''
                />
                {colorFieldsets.map((colorFieldset, index) => {
                    return <FormFieldset key={index} className='NPFImages__fieldset'>
                        <FormPromptWithSub prompt={colorFieldset.name} promptSubtitle='' />
                        <FormUrlInput 
                            id={'color-swatch-url-' + index}
                            name='swatchUrl'
                            prompt='Color swatch URL'
                            currentValue={colorFieldset.swatchUrl}
                            handleChange={(event) => {
                                changeSwatchInput(event, index)
                            }}
                        />
                        
                        {colorFieldset.imageUrls.map((colorImageUrl, i) => (
                            <div key={i} className='NewProductForm__image-field'>
                                <button 
                                    className='NewProductForm__remove-image'
                                    type='button'
                                    onClick={() => removeImage(index, i)}
                                >
                                    REMOVE
                                </button>

                                <FormUrlInput
                                    id={'color-image-urls-' + i}
                                    name='imageUrls'
                                    prompt='Image URL for this color'
                                    currentValue={colorImageUrl}
                                    handleChange={event => {
                                        changeImageInput(index, i, event)
                                    }}
                                />
                            </div>
                        ))}
                        
                        <FormButton
                            buttonText='ADD AN IMAGE FOR THIS COLOR'
                            handleClick={() => addImage(index)}
                        />
                   </ FormFieldset>
                })}
            </FormPage>

            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => { setPage(currentPage - 1) }} 
                nextButton={() => { nextButton() }}
            />
        </div>
    )
}

NPFImages.defaultProps = {
    colorFieldsets: [
        { 
            name: '', 
            descriptionId: '1' ,
            swatchUrl: '',
            imageUrls: ['']
        }
    ],
    currentPage: 0,
    setPage: () => {},
    setColorFieldsets: () => {}
}

export default NPFImages