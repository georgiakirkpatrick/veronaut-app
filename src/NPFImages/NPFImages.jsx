import React, { useState } from 'react'
import '@fortawesome/react-fontawesome'
import FormPage from '../FormPage/FormPage'
import FormButton from '../FormButton/FormButton'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import NPFFooter from '../NPFFooter/NPFFooter'
import './NPFImages.css'

const NPFImages = props => {
  const {
    colorFieldsets,
    currentPage,
    setColorFieldsets,
    setPage
  } = props

  const [images, setImages] = useState(false)

  const addImage = (index) => {
    const fieldsetToUpdate = colorFieldsets[index]
    fieldsetToUpdate.imageUrls.push('')
    setColorFieldsets([ ...colorFieldsets ])
  }

  const changeImageInput = (index, i, event) => {
    const values = [...colorFieldsets]
    values[index][event.target.name][i] = event.target.value

    setImages(values[index].imageUrls + 1)
  }

  const nextButton = () => {
    const missingUrls = []

    colorFieldsets.forEach(fieldset => {
      fieldset.imageUrls.forEach(field => {
        if (field === '' || null) {
          missingUrls.push(
            field.name
          )
        }
      })
    })
      
    if (missingUrls.length >= 1) {
      alert(`Please enter at least one image url for each color and remove any empty fields.  Having trouble getting an image URL?  Just remove any empty fields and click 'Next'.`)
    } else {
      setPage(currentPage + 1)
    }
  }

  const removeImage = (index, i) => {
    const newColors = [...colorFieldsets]
    newColors[index].imageUrls.splice(i, 1)

    setColorFieldsets(newColors)
  }

  return (
    <div className='NPFImages'>
      <FormPage title='Images'>
        <FormPromptWithSub 
          prompt='Enter the image URLs of up to four images for each color option, one URL per line.'
          promptSubtitle={`To copy an image URL from a product page, follow these steps: on a computer, right-click on the image, then select 'Copy Image Address'.  
          On a mobile device, tap and hold the image, then select 'Copy'.  If 'Copy' is not an option, select 'Share...', then 'Copy'.
          
          Remove any empty fields before tapping 'Next'.`}
        />
          {colorFieldsets.map((colorFieldset, index) => {
            const formattedColor = colorFieldset.name.length > 2
                ? colorFieldset.name[0].toUpperCase() + colorFieldset.name.substr(1)
                : 'Color ' + index

            return <FormFieldset key={index}>
              <FormPromptWithSub
                  prompt={formattedColor}
                  promptSubtitle={`Enter the image URLs of up to four product images in ${formattedColor}, one URL per line.`}
              />
                {colorFieldset.imageUrls.map((colorImageUrl, i) => {
                  return <div key={i} className='NPFImages__image-field'>
                    <button 
                      className='NPFImages__remove-image'
                      type='button'
                      onClick={() => removeImage(index, i)}
                    >
                      REMOVE
                    </button>

                    <FormUrlInput
                      id={'color-image-urls-' + i}
                      name='imageUrls'
                      prompt='Image URL for this color'
                      currentValue={colorImageUrl}
                      handleChange={event => {
                          changeImageInput(index, i, event)
                      }}
                    />
                  </div>
                })}

                <FormButton
                  buttonText='Add an image for this color'
                  handleClick={() => addImage(index)}
                />

                {images 
                  ? <div>
                    <p>Image preview</p>

                    <ul className='NPFImages__preview-images'>
                      {
                        colorFieldset.imageUrls.map((colorImageUrl, i) => (
                          colorImageUrl.length > 3
                            ? <li key={index + '-' + i} className='NPFImages__preview-img'>
                              <img
                                src={colorImageUrl}
                                alt={colorFieldset.name + '-image-' + i}
                              />
                            </li>
                            : <li key={index + '-' + i} className='NPFImages__preview-img'>
                              <br/>
                            </li>
                        ))
                      }
                    </ul>
                  </div>
                  : <br />
                }
            </ FormFieldset>
          })}
      </FormPage>

      <NPFFooter 
          buttons='prevNext' 
          previousButton={() => { setPage(currentPage - 1) }} 
          nextButton={() => { nextButton() }}
      />
    </div>
  )
}

NPFImages.defaultProps = {
  colorFieldsets: [
    { 
      name: 'test name', 
      descriptionId: '1' ,
      imageUrls: ['']
    }
  ],
  currentPage: 0,
  setPage: () => {},
  setColorFieldsets: () => {}
}

export default NPFImages