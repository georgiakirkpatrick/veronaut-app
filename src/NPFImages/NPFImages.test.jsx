import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFImages from './NPFImages'

describe('NPFImages', () => {
  it ('renders without crashing', () => {
    render(<NPFImages />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<NPFImages />)
    expect(tree).toMatchSnapshot()
  })
})