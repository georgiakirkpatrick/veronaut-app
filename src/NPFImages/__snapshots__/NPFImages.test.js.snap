// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFImages render the UI as expected 1`] = `
<div
  className="NPFImages"
>
  <section
    className="FormPage"
  >
    <h1>
      Images
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Enter the image URLs of up to four images for each color option, one URL per line.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          To copy an image URL from a product page, follow these steps: on a computer, right-click on the image, then select 'Copy Image Address'.  
                    On a mobile device, tap and hold the image, then select 'Copy'.  If 'Copy' is not an option, select 'Share...', then 'Copy'.
                    
                    Remove any empty fields before tapping 'Next'.
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <div />
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Test name
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            Enter the image URLs of up to four product images in test name, one URL per line.
          </p>
        </div>
        <div
          className="NPFImages__image-field"
        >
          <button
            className="NPFImages__remove-image"
            onClick={[Function]}
            type="button"
          >
            REMOVE
          </button>
          <div
            className="FormUrlInput"
          >
            <label
              htmlFor="color-image-urls-0"
            >
              Image URL for this color
            </label>
            <input
              id="color-image-urls-0"
              name="imageUrls"
              onChange={[Function]}
              required={true}
              type="url"
              value=""
            />
          </div>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          Add an image for this color
        </button>
        <br />
      </fieldset>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
